src/libasr/codegen/asr_to_c_cpp.h

```diff
@@ -1137,6 +1137,45 @@ R"(#include <stdio.h>
         last_expr_precedence = 16;
     }
 
+    void visit_StringConcat(const ASR::StringConcat_t &x) {
+    self().visit_expr(*x.m_left);
+    std::string left_expr = src;
+     self().visit_expr(*x.m_right);
+    std::string right_expr = src;
```

Comment:
For `StringConcat`, `lfortran_intrinsic.c` already has a function `_lfortran_strcat`. You can directly use that.